Лабораторная работа 3/task3.3.py

```diff
@@ -0,0 +1,60 @@
+# TODO  Напишите функцию count_letters
+def count_letters(main_str):
+    letter_count = dict()
+    for char in main_str.lower():
+        if char.isalpha():
+            if char in letter_count:
+                letter_count[char] += 1
+            else:
+                letter_count[char] = 1
+                return letter_count
+
+# TODO Напишите функцию calculate_frequency
+def calculate_frequency(letter_count):
+    total_letters = sum(letter_count.values())
+    frequency = dict()
+    for letter, count in letter_count.items():
+        frequency[letter] = round(count / total_letters, 2)
+    return frequency
```

Comment:
Неплохо!
Есть небольшая ошибка.
Сейчас Ваш код выводит лишь одну букву и ее частоту появления.
Ошибка находится в функции `count_letters`. В цикле for ты возвращаешь словарь `letter_count `после первого же символа, который является буквой. Это происходит потому, что условие return `letter_count `находится внутри цикла `for`, а должно находиться за пределами этого цикла.
Например, Вашу функцию можно исправить вот так:
```suggestion
def count_letters(main_str):
    letter_count = dict()
    for char in main_str.lower():
        if char.isalpha():
            if char in letter_count:
                letter_count[char] += 1
            else:
                letter_count[char] = 1
    # Возвращаем результат вне цикла
    return letter_count
```